content/topics/unit-testing/tdd-horrors/_index.md

```diff
@@ -0,0 +1,214 @@
+---
+title: TDD Horrors
+ready: true
+---
+
+# Common TDD pitfalls.
+These are all the most common TDD problems recruits tend to come across, read and make sure you didn't make any of these or any mistakes at all.
+
+- **Write tests**. Recruits in the past have done the mistake of not writing tests for TDD assignments, make sure you avoid this weird mistake. always write tests for your TDD projects.
+- No code in your TDD file, This is a repetition of the first point, but it's worth it. write tests people.
+- Files naming convention, name your files properly and according to what's inside them. E.g. if you have a file called add.js that only contains a multiply function then something is wrong. Do this:
+<!-- examples -->
+
+**Python**:
+
+`add.py`
+```py
+def add(***):
+  # some operations
+```
+**Javascript**
+
+`add.js`
+```js
+function add(***){
+  // operations
+}
+```
+
+- Defining a test case(testing function) inside a function. That will pass, even when it's supposed to fail.
+
+---
+**Python example:**
+
+```py
+def testOne():
+
+  def testSomething():
+    assert add("1,23,5") == 25
+
+  def testSomething():
+    assert add("1,23,5") == 29
+
+```
+instead you could put your tests into classes as methods, like so:
+
```

Comment:
Or you could just follow the principle of KISS and "flat is better than nested"  and just do the simple flat thing

```
def totally_valid_test():
    etc

def another_test():
   woo
```